kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/KubernetesHttpClientFactory.java

```diff
@@ -0,0 +1,144 @@
+        this.kubernetesPrivateKeyLoader = kubernetesPrivateKeyLoader;
+        this.resourceResolver = resourceResolver;
+        this.defaultHttpClientFilterResolver = defaultHttpClientFilterResolver;
+    }
+
+    @Singleton
+    @Named(CLIENT_ID)
+    @BootstrapContextCompatible
+    protected DefaultHttpClient getKubernetesHttpClient() {
+        URI uri = URI.create(kubeConfig.getCluster().server());
+
+        return new DefaultHttpClient(LoadBalancer.fixed(uri),
+            null,
+            new DefaultHttpClientConfiguration(),
+            null,
+            defaultHttpClientFilterResolver,
+            defaultHttpClientFilterResolver.resolveFilterEntries(new ClientFilterResolutionContext(Collections.singletonList("kubernetes-client"), null)),
+            new DefaultThreadFactory(MultithreadEventLoopGroup.class),
+            new KubernetesClientSslBuilder(resourceResolver, kubeConfig, kubernetesPrivateKeyLoader),
+            createDefaultMediaTypeRegistry(),
+            createDefaultMessageBodyHandlerRegistry(),
+            WebSocketBeanRegistry.EMPTY,
+            new DefaultRequestBinderRegistry(ConversionService.SHARED),
+            null,
+            NioSocketChannel::new,
+            NioDatagramChannel::new,
+            new NettyClientCustomizer() {
+                @Override
+                public @NonNull NettyClientCustomizer specializeForChannel(@NonNull Channel channel, @NonNull ChannelRole role) {
+                    return NettyClientCustomizer.super.specializeForChannel(channel, role);
+                }
+            },
+            null,
+            ConversionService.SHARED,
+            null);
+    }
+
+    private static MediaTypeCodecRegistry createDefaultMediaTypeRegistry() {
+        JsonMapper mapper = JsonMapper.createDefault();
+        ApplicationConfiguration configuration = new ApplicationConfiguration();
+        return MediaTypeCodecRegistry.of(
+            new JsonMediaTypeCodec(mapper, configuration, null),
+            new JsonStreamMediaTypeCodec(mapper, configuration, null)
+        );
+    }
+
+    private static MessageBodyHandlerRegistry createDefaultMessageBodyHandlerRegistry() {
+        ApplicationConfiguration applicationConfiguration = new ApplicationConfiguration();
+        ContextlessMessageBodyHandlerRegistry registry = new ContextlessMessageBodyHandlerRegistry(
+            applicationConfiguration,
+            NettyByteBufferFactory.DEFAULT,
+            new NettyByteBufMessageBodyHandler(),
+            new NettyWritableBodyWriter(applicationConfiguration)
+        );
+        JsonMapper mapper = JsonMapper.createDefault();
+        registry.add(MediaType.APPLICATION_JSON_TYPE, new NettyJsonHandler<>(mapper));
+        registry.add(MediaType.APPLICATION_JSON_TYPE, new NettyCharSequenceBodyWriter());
+        registry.add(MediaType.APPLICATION_JSON_STREAM_TYPE, new NettyJsonStreamHandler<>(mapper));
+        return registry;
+    }
```

Comment:
would like to see how we can avoid this logic and recreating all these objects? //cc @yawkat

Reply:
All those will be removed once micronaut-core 4.7.0 gets released since that version contain http client builder implementation